code/modules/power/cable.dm

```diff
@@ -415,3 +412,172 @@
 		logTheThing(LOG_STATION, user, "lays a cable[powered == 1 ? " (powered when connected)" : ""] at [log_loc(src)].")
 
 	return
+
+/// a cable spawner which can spawn multiple cables to connect to other cables around it.
+/obj/cablespawner
+	level = 1
+	name = "power cable spawner"
+	desc = "An item that should spawn actual cables. If you're reading this, something's wrong."
+	icon = 'icons/obj/power_cond.dmi'
+	icon_state = "superstate"
+	var/iconmod = null
+	layer = CABLE_LAYER
+	plane = PLANE_NOSHADOW_BELOW
+	color = "#DD0000"
+	// some bit flags for 8 bit directions
+	var/const/SW = 128
+	var/const/NW = 64
+	var/const/SE = 32
+	var/const/NE = 16
+	/// this 8 bit variable uses bit flags to tell which 8 tiles need connections
+	var/cable_surr = 0
+	/*
+	* bitflag of the tiles surrounding itself:
+	* i.e.(10)(9)(6)(5) 8  4  2  1 (normal bitflags) (used when you only need one direction)
+	* i.e. SW NW SE NE  W  E  S  N (corresponding directions)
+	*     128 64 32 16  8  4  2  1 (actual bitflags) (used for multiple directions)
+	* i think someone called it 8 bit directions once? idk
+	* anyway thats what cable_surr does
+	*/
+
+/// creates new cablespawners, which should replace themselves
+/obj/cablespawner/New()
+	..()
+/obj/cablespawner/initialize()
+	. = ..()
+	src.check()
+	src.replace()
+/// if there is only cablespawners, not all 8 directions are needed. This removes some flotsam.
+/obj/cablespawner/proc/optimise()
+	/*
+	Diagonals are ugly. So if the option to connect to a diagonal tile orthogonally presents itself
+	we'll get rid of the corners and connect in NESW directions first.
+	This gets rid of diagonals in 2x2 and 3x3 grids, and stops small 'L's from becoming triangles.
+	if a diagonal tile is next to a cardinal, we disregard it.
+	This won't work on the manually connected cables.
+	*/
+	if (cable_surr & NE)
+		if (cable_surr & NORTH || cable_surr & EAST)
+			cable_surr &= ~NE
+	if (cable_surr & NW)
+		if (cable_surr & NORTH || cable_surr & WEST)
+			cable_surr &= ~NW
+	if (cable_surr & SE)
+		if (cable_surr & SOUTH || cable_surr & EAST)
+			cable_surr &= ~SE
+	if (cable_surr & SW)
+		if (cable_surr & SOUTH || cable_surr & WEST)
+			cable_surr &= ~SW
+
+/// checks around itself for cables, adds up to 8 bits to cable_surr
+/obj/cablespawner/proc/check(var/obj/cable/cable)
+	for (var/obj/cablespawner/spawner in orange(1, src))
+	// cablespawners around itself
+		var/tempflag = 0
```

Comment:
should be temp_flags - this is what all the automation is hung up on

Reply:
yeah i noticed this a bit late. A little bit embarrassing, haha